pkg/cmd/api/api.go

```diff
@@ -431,14 +455,18 @@ func processResponse(resp *http.Response, opts *ApiOptions, bodyWriter, headersW
 	} else if isJSON && opts.IO.ColorEnabled() {
 		err = jsoncolor.Write(bodyWriter, responseBody, "  ")
 	} else {
-		if isJSON && opts.Paginate && !isGraphQLPaginate && !opts.ShowResponseHeaders {
-			responseBody = &paginatedArrayReader{
-				Reader:      responseBody,
-				isFirstPage: isFirstPage,
-				isLastPage:  isLastPage,
-			}
+		if isJSON && opts.merger != nil && !opts.ShowResponseHeaders {
+			responseBody = opts.merger.NewPage(responseBody, isLastPage)
 		}
```

Comment:
**How is pagination being handled for REST requests that don't use the new merge pages feature?**

I'll admit the changes are a little tricky for me to wrap my head around, but it seems previously we would fall into this conditional case for REST pagination calls.  Reviewing the rest of the changes, it seems we only have a `merger` if the new flag is used.

Reply:
Yes, we only have a `merger` is paging is requested, but this particular block was there before saying that, if headers are requested, we keep them interstitial which means you get multiple pages of a single array each. This was how Nate's changes worked to preserve - out of fear of what we don't know, in terms of how people use this - compatibility.

Reply:
Created a `NopMerger` per recommendation below, following the pattern set by `io.NopCloser`. Hopefully simplifies things conceptually a little, though conditions didn't change much.